spyder/plugins/ipythonconsole/widgets/tests/test_kernelconnect.py

```diff
@@ -0,0 +1,201 @@
+# Fixtures
+# =============================================================================
+@pytest.fixture
+def connection_dialog_factory(qtbot, request):
+    """Set up kernel connection dialog."""
+    class DialogFactory(object):
+        def get_default_dialog(self):
+            dialog = KernelConnectionDialog()
+            request.addfinalizer(dialog.close)
+            return dialog
+
+        def submit_filled_dialog(self, use_keyfile, save_settings):
+            dlg = self.get_default_dialog()
+
+            # fill out cf path
+            dlg.cf.clear()
+            qtbot.keyClicks(dlg.cf, cf_path)
+            # check the remote kernel box
+            dlg.rm_group.setChecked(True)
+            # fill out hostname
+            dlg.hn.clear()
+            qtbot.keyClicks(dlg.hn, hn)
+            # fill out port
+            dlg.pn.clear()
+            qtbot.keyClicks(dlg.pn, str(pn))
+            # fill out username
+            dlg.un.clear()
+            qtbot.keyClicks(dlg.un, un)
+
+            if use_keyfile:
+                # select ssh keyfile radio
+                dlg.kf_radio.setChecked(True)
+                assert dlg.kf.isEnabled()
+
+                # fill out ssh keyfile
+                dlg.kf.clear()
+                qtbot.keyClicks(dlg.kf, kf)
+
+                # fill out passphrase
+                dlg.kfp.clear()
+                qtbot.keyClicks(dlg.kfp, kfp)
+            else:
+                # select password radio
+                dlg.pw_radio.setChecked(True)
+                assert dlg.pw.isEnabled()
+
+                # fill out password
+                dlg.pw.clear()
+                qtbot.keyClicks(dlg.pw, pw)
+
+            # check save connection settings
+            dlg.save_layout.setChecked(save_settings)
+
+            # Press ok and save connection settings
+            qtbot.mouseClick(dlg.accept_btns.button(QDialogButtonBox.Ok),
+                             Qt.LeftButton)
+
+            return dlg
+
+    def teardown():
+        """Clear existing-kernel config and keyring passwords"""
+        CONF.remove_section("existing-kernel")
+
+        try:
+            import keyring
+            keyring.set_password("existing_kernel",
+                                 "ssh_key_passphrase", "")
+            keyring.set_password("existing_kernel",
+                                 "ssh_password", "")
+        except Exception:
+            pass
+
+    request.addfinalizer(teardown)
+    return DialogFactory()
+
+
+cf_path = "test_cf_path"
+un = "test_username"
+hn = "test_hostname"
+pn = 123
+kf = "test_kf"
+kfp = "test_kfp"
+pw = "test_pw"
+
+
+# =============================================================================
+# Tests
+# =============================================================================
+def test_kernel_connection_dialog_remember_input_ssh_passphrase(
+        qtbot, connection_dialog_factory):
+    """Test that the dialog remember user's kernel connection
```

Comment:
*remembers *the user's . Also, by our convention, multi-line docstrings should have the ``"""`` on separate lines from the text (i.e.

```
"""
Test that blah blah
blah blah blah blah.
"""
```